### PR TITLE
S1 timing splines implementation for top/bottom arrays

### PR DESCRIPTION
**Pull request by terliuk**
**What is the problem / what does the code in this PR do**
This code implements splines for timing distribution of S1 photon hits for each array, as was discussed in recent MC telecon.

**WFSim needs you:**
 - Please add a test for this PR, as a bare minimum, make sure it's covered in coveralls!

 - Can you add a docsting to all your functions?
I added comments describing the logic inside the function

### CODE REVIEW

**Comment by petergaemers** [comment]
Hi Andrii, thanks a lot. I have a question. So where is the scintillation delay? It looks like it's already included in the spline?

**Comment by terliuk** [comment]
Yes, currently it is included in the spline. In my plans is also to provide propagation time + NEST implementation, but it might need a bit more time to fully implement. Current spline is Geant4 propagation + 32.1 keV Kr83m NEST.

File: wfsim/core.py [hunk 1]
```diff
@@ -373,7 +373,7 @@ def photon_channels(positions, n_photons, config, s1_pattern_map):
         return _photon_channels
 
     @staticmethod
-    def photon_timings(t, n_photons, recoil_type, config, phase):
+    def photon_timings(t, n_photons, recoil_type, config, phase, channels=None,positions=None,resource=None):
```
Comment (ramirezdiego) [inline wfsim/core.py:376]: (Small comment) Since we're here, would be nice to extend a bit on the args description in the docstring below, plus of course add the new args you coded.

Comment (terliuk) [reply wfsim/core.py:376]: Done

**Review by petergaemers — APPROVED** [review]
